src/dictionary/manager.ts

```diff
@@ -294,19 +294,23 @@ class SchemaManager {
   };
 }
 
-export const revalidateAllDonorClinicalEntitiesAgainstSchema = (
+export const revalidateAllDonorClinicalEntitiesAgainstSchema = async (
   donor: DeepReadonly<Donor>,
   schema: dictionaryEntities.SchemasDictionary,
 ) => {
   const clinicalSchemaNames = getSchemaNamesForDonorClinicalEntities(donor);
   let isValid = true;
-  clinicalSchemaNames.forEach((schemaName: ClinicalEntitySchemaNames) => {
+  for (const schemaName of clinicalSchemaNames) {
     if (!isValid) {
       return;
     }
-    const errs = MigrationManager.validateDonorEntityAgainstNewSchema(schemaName, schema, donor);
-    isValid = !errs || errs.length == 0;
-  });
+    const errs = await MigrationManager.validateDonorEntityAgainstNewSchema(
+      schemaName,
+      schema,
+      donor,
+    );
+    isValid = isValid && (!errs || errs.length == 0);
```

Comment:
`isValid` is always going to be `true` here, so the original line is correct

Reply:
Right, cleaned up